scalyr/opentelemetry-exporter-dataset
# Upgrade to OTEL Contrib 0.104.0

It used to work - run 479 - https://github.com/scalyr/opentelemetry-exporter-dataset/actions/runs/9493515115/job/26162336976
And then it stopped working - run 480 - https://github.com/scalyr/opentelemetry-exporter-dataset/actions/runs/9510451598

```
#16 330.8 Error: failed to compile the OpenTelemetry Collector distribution: go subcommand failed with args '[build -trimpath -o otelcol-contrib -ldflags=-s -w]': exit status 1, error message: # github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator
#16 330.8 /go/pkg/mod/github.com/open-telemetry/opentelemetry-collector-contrib/receiver/receivercreator@v0.101.0/runner.go:136:22: undefined: component.UnmarshalConfig
#16 330.8 # go.opentelemetry.io/collector/otelcol/internal/configunmarshaler
#16 330.8 /go/pkg/mod/go.opentelemetry.io/collector/otelcol@v0.101.0/internal/configunmarshaler/configs.go:46:23: undefined: component.UnmarshalConfig
#16 330.8 
#16 ERROR: process "/bin/sh -c CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} $GOPATH/bin/builder --config=/build/otelcol-builder.datasetexporter-latest.yaml" did not complete successfully: exit code: 1
```

[logs_24894699041.zip](https://github.com/user-attachments/files/16086380/logs_24894699041.zip)


I have added following instructions how to upgrade into README

## How To Upgrade

1. Update https://github.com/scalyr/opentelemetry-collector-contrib
   1. ```bash
      cd ../opentelemetry-collector-contrib
      # checkout main branch
      git checkout main
      # sync with upstream
      gh repo sync scalyr/opentelemetry-collector-contrib -b main
      # pull changes
      git pull
      # update dataset-latest branch
      git checkout dataset-latest
      # merge main
      git pull
      git merge main
      # push changes
      git push
      ```
   * Check in the UI, that the branches are not behind
     * https://github.com/scalyr/opentelemetry-collector-contrib/tree/main
     * https://github.com/scalyr/opentelemetry-collector-contrib/tree/datasetexporter-latest
   * If they are behind use `Sync fork` button in the UI to sync them
2. Create new branch for the new version
   ```bash
   git checkout -b DPDV-6415-update-packages
   ```
3. Pull all subtress
   ```bash
   make subtrees-pull
   ```
4. Push changes
   ```bash
   git push
    ```
5. Update docker files and configurations to use the new version of the collector
   * old version is from here - [otelcol-builder.yaml](https://github.com/scalyr/opentelemetry-exporter-dataset/blob/main/otelcol-builder.yaml)
   * new version is from the [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases)
   ```bash
   ./scripts/update-otel-version.sh -f v0.101.0 -t v0.104.0
   ```
6. Update collector configuration to use the new version of dataset-go
   * old version is from here - [otelcol-builder.yaml](https://github.com/scalyr/opentelemetry-exporter-dataset/blob/main/otelcol-builder.yaml)
   * new version is from the [dataset-go](https://github.com/scalyr/dataset-go/releases)
   ```bash
   ./scripts/update-dataset-go-version.sh -f v0.18.0 -t v0.20.0
   ```
7. Build all docker images
   ```bash
   make docker-build
   ```
8. Run e2e tests
   * Set environment variables for the tests
     ```bash
     export TEST_RUN_SERVERHOST=`date +"%s"`
     export DATASET_URL=https://app.scalyr.com/
     export DATASET_API_KEY=FOO
     ```
   * Run the tests
     ```bash
     make test-e2e
     ```
